eth/downloader/queue.go

```diff
@@ -86,81 +131,141 @@ func (q *queue) Reset() {
 	q.lock.Lock()
 	defer q.lock.Unlock()
 
+	q.mode = FullSync
+	q.fastSyncPivot = 0
+
 	q.hashPool = make(map[common.Hash]int)
 	q.hashQueue.Reset()
 	q.hashCounter = 0
 
-	q.headerPool = make(map[common.Hash]*types.Header)
-	q.headerQueue.Reset()
 	q.headerHead = common.Hash{}
 
-	q.pendPool = make(map[string]*fetchRequest)
+	q.blockTaskPool = make(map[common.Hash]*types.Header)
+	q.blockTaskQueue.Reset()
+	q.blockPendPool = make(map[string]*fetchRequest)
+	q.blockDonePool = make(map[common.Hash]struct{})
+
+	q.receiptTaskPool = make(map[common.Hash]*types.Header)
+	q.receiptTaskQueue.Reset()
+	q.receiptPendPool = make(map[string]*fetchRequest)
+	q.receiptDonePool = make(map[common.Hash]struct{})
+
+	q.stateTaskIndex = 0
+	q.stateTaskPool = make(map[common.Hash]int)
+	q.stateTaskQueue.Reset()
+	q.statePendPool = make(map[string]*fetchRequest)
+	q.stateScheduler = nil
+
```

Comment:
OK, makes sense. Just wanted to make a sanity check from diffing struct fields and the ones being reset.